common/src/main/java/com/wynntils/models/abilities/bossbars/OphanimBar.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.abilities.bossbars;
+
+import com.wynntils.handlers.bossbar.TrackedBar;
+import com.wynntils.models.abilities.type.OphanimOrb;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OphanimBar extends TrackedBar {
+    // Test in OphanimBar_OPHANIM_PATTERN
+    private static final Pattern OPHANIM_PATTERN =
+            Pattern.compile("^§710s Healed: §f(?<healed>\\d+)% §[3468]\\[(?<orbs>((§[bce7])?⏺){0,6})(§[3468])?\\]$");
+
+    private static final Pattern ORB_PATTERN = Pattern.compile("(?<color>§[bce7])?⏺");
+
+    private int healed = 0;
+    private final List<OphanimOrb> orbs = new ArrayList<>();
+
+    public OphanimBar() {
+        super(OPHANIM_PATTERN);
+    }
+
+    public int getHealed() {
+        return healed;
+    }
+
+    public List<OphanimOrb> getOrbs() {
+        return Collections.unmodifiableList(orbs);
+    }
+
+    @Override
+    public void onUpdateName(Matcher match) {
+        healed = Integer.parseInt(match.group("healed"));
+
+        orbs.clear();
+        Matcher orbMatcher = ORB_PATTERN.matcher(match.group("orbs"));
+
+        OphanimOrb.HealthState healthState = null;
+        int start = 0;
+        while (orbMatcher.find(start)) {
+            String colorCode = orbMatcher.group("color");
+            if (colorCode != null) {
+                healthState = OphanimOrb.HealthState.fromColorCode(colorCode);
+            }
+
+            orbs.add(new OphanimOrb(healthState));
```

Comment:
What if the color code was null? `healthState` is then either null or in the state if the last orb. I think you need to handle that. If the regex "doesn't allow" colorCode to be null, just add a comment or a `WynntilsMod.warn` call to make sure we prevent bugs, if anything get changed.

Reply:
We expect the color code to not exist sometimes.

This has to do with `Component` to `StyledText` conversion. Wynncraft sends a color code before every orb icon, but the conversion to StyledText removes redundant code formatting. So if three consecutive Orbs are of red color for example, only the first one has a color code prepended. This logic works around that by caching the health state of the last orb and only if a new color code prefix is present inside the pattern update it.

So in theory the healthState should not be null as we expect the first orb to have a color code prefix.

Will definitely add a warning and maybe some explanatory comments as well

Reply:
> Will definitely add a warning and maybe some explanatory comments as well

This is the most important part :). I know StyledText, in fact, I wrote it, but still, one can't comprehend all of it's "side-effects" when reading code using it, like here.